server/src/main/java/org/candlepin/model/AsyncJobStatus.java

```diff
@@ -101,22 +101,31 @@ public boolean isTerminal() {
     @NotNull
     private String name;
     private String origin;
+    private String executor;
     private String principal;
 
-    // TODO: If we add more stuff like this field, we should create another table to store
-    // job metadata rather than continuing to bolt more columns onto this for fields that may
-    // not be filled/configured globally (i.e. announce completion and job log level)
-    @Column(name = "correlation_id")
-    private String correlationId;
+    @Column(name = "log_level")
+    private String logLevel;
+    @Column(name = "log_execution_details")
+    private boolean logExecutionDetails;
+
+    @ElementCollection(fetch = FetchType.EAGER)
+    @CollectionTable(name = "cp2_async_job_metadata", joinColumns = @JoinColumn(name = "job_id"))
+    @MapKeyColumn(name = "key")
+    @Column(name = "value")
+    private Map<String, String> metadata;
+
+    @Column(name = "previous_state")
+    private JobState previousState;
```

Comment:
With the failure recovery bits, it's possible to skip states or otherwise get stuck if the right situation occurs. This is something that will be used chiefly for human debugging/identification purposes.